packages/core/src/asset/ResourceManager.ts

```diff
@@ -136,6 +144,23 @@ export class ResourceManager {
     return this._assetPool[instanceId];
   }
 
+  getResourceByRef<T>(ref: { refId: string; key?: string; isClone?: boolean }): Promise<T> {
```

Comment:
Add comments with `@beta  just for internal editor, not recommended for developers`